ipatests/test_integration/test_installation.py

```diff
@@ -847,3 +847,43 @@ def test_replicamanage_del(self):
 
         assert_text = 'Deleted IPA server "%s"' % self.replicas[0].hostname
         assert assert_text in cmd_output.stdout_text
+
+
+class TestInstallReplicaAgainstSpecificServer(IntegrationTest):
+    """Installation of replica against a specific server
+
+    Test to check replica install against specific server. It uses master and
+    replica1 without CA and having custodia service stopped. Then try to
+    install replica2 from replica1 and expect it to get fail as specified
+    server is not providing all the services.
+
+    related ticket: https://pagure.io/freeipa/issue/7566
+    """
+
+    num_replicas = 2
+
+    @pytest.mark.xfail(reason='freeipa ticket 7566', strict=True)
+    def test_replica_install_against_server(self):
+        tasks.install_master(self.master)
+
+        # install replica1 without CA
+        cmd = tasks.install_replica(self.master, self.replicas[0],
+                                    setup_ca=False, setup_dns=True,
+                                    promote=False)
+
+        # check for warning that CA is not installed on server
+        warn = 'WARNING: The CA service is only installed on one server'
+        assert warn in cmd.stderr_text
+
+        # stop custodia service on replica1
+        self.replicas[0].run_command('systemctl stop ipa-custodia.service')
+
+        # check if custodia service is stopped
+        cmd = self.replicas[0].run_command('ipactl status')
+        assert 'ipa-custodia Service: STOPPED' in cmd.stdout_text
+
+        # install replica2 against replica1, as custodia service is stopped
+        # on replica1, installation on replica2 should fail
+        cmd = tasks.install_replica(self.replicas[0], self.replicas[1],
+                                    promote=False, raiseonerr=False)
+        assert cmd.returncode != 0
```

Comment:
With PR #3664, `ipa-replica-install --server ...` will return 4 when the provided remote master does not have CA/KRA. You can also check for the error message `please provide a server with the CA role`.

You can also enhance this scenario:
- install KRA on the first master (add the param `setup_kra=True` to `tasks.install_master` on line 867
- add a new test function that will install the CA on replicas[0], and try to call `ipa-replica-install --setup-ca --setup-kra --server replicas[0]` on replicas[1], should be refused because KRA not present on replicas[0] and produce error message `please provide a server with the KRA role`
- add a new test function that calls `ipa-replica-install --server master --setup-ca --setup-kra`, should succeed. Check that the replica agreement are established with master only.